crates/rlp/src/error.rs

```diff
@@ -12,6 +12,8 @@ pub enum Error {
     LeadingZero,
     /// Overran input while decoding.
     InputTooShort,
+    /// Additional trailing bytes found after decoding.
+    TrailingBytes,
```

Comment:
I think this can be expressed with the Unexptectedlength variant

Reply:
Considering that this is currently returned when the payload length doesn't match the array size or other payload length issues, I didn't think it would be a good fit for this since `decode_exact()` doesn't look at the payload, but rather the entire RLP encoded data. However, if this is just me, I can definitely adjust `UnexpectedLength' to better account for both cases...

Reply:
I would be ok with the new variant however this is unfortunately a breaking change because the enum is not marked as `#[non_exhaustive]`. So for now I would prefer the other option and adding a TODO for the new variant